python/tskit/trees.py

```diff
@@ -2917,6 +2918,68 @@ def samples(self, population=None, population_id=None):
             samples = samples[sample_population == population]
         return samples
 
+    def write_fasta(self, output, sequence_ids=None, wrap_width=60):
+        """
+        Writes a fasta formatted file from tree sequence data to the
+        specified file-like object.
+
+        Default `sequence_ids` (i.e. the text immediately following ">") are
+        "tsk_{sample_number}" e.g. "tsk_0", "tsk_1" etc. They can be set by providing
+        a list of strings to the `sequence_ids` argument, which must equal the length
+        of the number of samples. Please ensure that these are unique and compatible with
+        fasta standards, since we do not check this.
+        Default `wrap_width` for sequences is 60 characters in accordance with fasta
+        standard outputs, but this can be specified. In order to avoid any line-wrapping
+        of sequences, set `wrap_width = 0`.
+
+        Example usage:
+
+        .. code-block:: python
+            with open("output.fasta", "w") as fasta_file:
+                ts.write_fasta(fasta_file)
+
+        This can also be achieved on the command line use the ``tskit fasta`` command,
+        e.g.:
+
+        .. code-block:: bash
+
+            $ tskit fasta example.trees > example.fasta
+
+        :param File output: The file-like object to write the fasta output.
+        :param list(str) sequence_ids: A list of string names to uniquely identify
+            each of the sequences in the fasta file. If specified, this must be a
+            list of strings of length equal to the number of samples which are output.
+            Note that we do not check the form of these strings in any way, so that it
+            is possible to output bad fasta IDs (for example, by including spaces
+            before the unique identifying part of the string).
+            The default is to output ``tsk_j`` for the jth individual.
+        :param int wrap_width: This parameter specifies the number of sequence
+            characters to include on each line in the fasta file, before wrapping
+            to the next line for each sequence. Defaults to 60 characters in
+            accordance with fasta standard outputs. To avoid any line-wrapping of
+            sequences, set `wrap_width = 0`. Otherwise, supply any positive integer.
+        """
+        # if not specified, IDs default to sample index
+        if sequence_ids is None:
+            sequence_ids = ["tsk_{}".format(j) for j in self.samples()]
+        if len(sequence_ids) != self.num_samples:
+            raise ValueError(
+                "sequence_ids must have length equal to the number of samples.")
+
+        wrap_width = int(wrap_width)
+        if wrap_width < 0:
+            raise ValueError("wrap_width must be a non-negative integer."
```

Comment:
Need some spaces at the end of these strings here (look at the string that is printed out, the words will be mushed together)